src/robot/ConcreteRobot.cpp

```diff
@@ -0,0 +1,451 @@
+{
+  using constraint::TestableIntersection;
+
+  if (mRootRobot != this)
+    return mRootRobot->getFullCollisionConstraint(
+        space, metaSkeleton, collisionFree);
+
+  auto selfCollisionFree = getSelfCollisionConstraint(space, metaSkeleton);
+
+  if (!collisionFree)
+    return selfCollisionFree;
+
+  // Make testable constraints for collision check
+  std::vector<TestablePtr> constraints;
+  constraints.reserve(2);
+  constraints.emplace_back(selfCollisionFree);
+  if (collisionFree)
+  {
+    if (collisionFree->getStateSpace() != space)
+    {
+      throw std::runtime_error("CollisionFree has incorrect statespace.");
+    }
+    constraints.emplace_back(collisionFree);
+  }
+
+  return std::make_shared<TestableIntersection>(space, constraints);
+}
+
+//==============================================================================
+TrajectoryPtr ConcreteRobot::planToConfiguration(
+    const MetaSkeletonStateSpacePtr& stateSpace,
+    const MetaSkeletonPtr& metaSkeleton,
+    const StateSpace::State* goalState,
+    const CollisionFreePtr& collisionFree,
+    double timelimit)
+{
+  auto collisionConstraint
+      = getFullCollisionConstraint(stateSpace, metaSkeleton, collisionFree);
+
+  return util::planToConfiguration(
+      stateSpace,
+      metaSkeleton,
+      goalState,
+      collisionConstraint,
+      cloneRNG().get(),
+      timelimit);
+}
+
+//==============================================================================
+TrajectoryPtr ConcreteRobot::planToConfiguration(
+    const MetaSkeletonStateSpacePtr& stateSpace,
+    const MetaSkeletonPtr& metaSkeleton,
+    const Eigen::VectorXd& goal,
+    const CollisionFreePtr& collisionFree,
+    double timelimit)
+{
+  auto goalState = stateSpace->createState();
+  stateSpace->convertPositionsToState(goal, goalState);
+
+  return planToConfiguration(
+      stateSpace, metaSkeleton, goalState, collisionFree, timelimit);
+}
+
+//==============================================================================
+TrajectoryPtr ConcreteRobot::planToConfigurations(
+    const MetaSkeletonStateSpacePtr& stateSpace,
+    const MetaSkeletonPtr& metaSkeleton,
+    const std::vector<StateSpace::State*>& goalStates,
+    const CollisionFreePtr& collisionFree,
+    double timelimit)
+{
+  return util::planToConfigurations(
+      stateSpace,
+      metaSkeleton,
+      goalStates,
+      collisionFree,
+      cloneRNG().get(),
+      timelimit);
+}
+
+//==============================================================================
+TrajectoryPtr ConcreteRobot::planToConfigurations(
+    const MetaSkeletonStateSpacePtr& stateSpace,
+    const MetaSkeletonPtr& metaSkeleton,
+    const std::vector<Eigen::VectorXd>& goals,
+    const CollisionFreePtr& collisionFree,
+    double timelimit)
+{
+  std::vector<StateSpace::State*> goalStates;
+  for (const auto goal : goals)
+  {
+    auto goalState = stateSpace->createState();
+    stateSpace->convertPositionsToState(goal, goalState);
```

Comment:
Is this actually setting `goalStates` somehow?

Reply:
Good catch! Now it does. ;)